adapter-base-quarkus/src/test/java/org/eclipse/hono/adapter/quarkus/KafkaRuntimeConfigProducerTest.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright (c) 2021 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+
+package org.eclipse.hono.adapter.quarkus;
+
+import javax.inject.Inject;
+
+import org.eclipse.hono.client.kafka.KafkaAdminClientConfigProperties;
+import org.eclipse.hono.client.kafka.KafkaProducerConfigProperties;
+import org.eclipse.hono.client.kafka.consumer.KafkaConsumerConfigProperties;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+import io.quarkus.test.junit.QuarkusTest;
+
+/**
+ * Verifies that {@link KafkaRuntimeConfigProducer} parses yaml files correctly and exposes the configuration as a
+ * beans.
+ */
+@QuarkusTest
+public class KafkaRuntimeConfigProducerTest {
+
+    @Inject
+    KafkaProducerConfigProperties kafkaProducerConfig;
+
+    @Inject
+    KafkaConsumerConfigProperties kafkaConsumerConfig;
+
+    @Inject
+    KafkaAdminClientConfigProperties kafkaAdminClientConfig;
+
+    /**
+     * Asserts that the Kafka configuration is exposed as a bean.
+     */
+    @Test
+    public void testThatBeanIsProvided() {
+        Assertions.assertNotNull(kafkaProducerConfig);
+        Assertions.assertNotNull(kafkaProducerConfig);
```

Comment:
any particular reason why we need to assert this twice?

Reply:
Please see #2560.